pkg/api/customization/alert/notifier_action.go

```diff
@@ -306,20 +312,39 @@ func testEmail(host, password, username string, port int, requireTLS bool, msg,
 		return fmt.Errorf("Failed to set recipient: %v", err)
 	}
 
-	if msg != "" {
-		// Data
-		w, err := c.Data()
-		if err != nil {
-			return err
-		}
-		defer w.Close()
+	wc, err := c.Data()
+	if err != nil {
+		return err
+	}
 
-		_, err = w.Write([]byte(msg))
-		if err != nil {
-			return err
-		}
+	defer wc.Close()
+
+	fmt.Fprintf(wc, "%s: %s\r\n", "From", sender)
+	fmt.Fprintf(wc, "%s: %s\r\n", "To", receiver)
+	fmt.Fprintf(wc, "%s: %s\r\n", "Subject", "Alert From Rancher: SMTP configuration validated")
+
+	buffer := &bytes.Buffer{}
+	multipartWriter := multipart.NewWriter(buffer)
+
+	fmt.Fprintf(wc, "Date: %s\r\n", time.Now().Format(time.RFC1123Z))
+	fmt.Fprintf(wc, "Content-Type: multipart/alternative;  boundary=%s\r\n", multipartWriter.Boundary())
+	fmt.Fprintf(wc, "MIME-Version: 1.0\r\n")
+
+	fmt.Fprintf(wc, "\r\n")
+
+	w, err := multipartWriter.CreatePart(textproto.MIMEHeader{"Content-Type": {"text/html; charset=UTF-8"}})
+	if err != nil {
+		return fmt.Errorf("Failed to send test email: %s", err)
 	}
 
+	_, err = w.Write([]byte(msg))
+	if err != nil {
+		return fmt.Errorf("Failed to send test email: %s", err)
+	}
+
+	multipartWriter.Close()
+	wc.Write(buffer.Bytes())
```

Comment:
is this function has error return?

Reply:
@zionwu looks like it does, and we should check for the error

Reply:
sure